pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -457,8 +540,11 @@ func GetPipelineConditionStatus(pr *v1beta1.PipelineRun, state PipelineRunState,
 			// If the pipeline is in stopping mode, all tasks that are not running
 			// already will be skipped. Otherwise these tasks end up in the
 			// incomplete count.
-			skipTasks++
-			withStatusTasks = append(withStatusTasks, rprt.PipelineTask.Name)
+			// this should never be the case for final task
+			if _, ok := dag.Nodes[rprt.PipelineTask.Name]; ok {
```

Comment:
this is used quite a few times, perhaps it's worth adding an helper, so it could be

```
case !rprt.IsStarted() && isStopping && !rprt.InDag(dag):
```

Reply:
tracking in #2852